### ISSUE TITLE
`name[prefix]` also does splitting when not enabled

### ISSUE THREAD

**Issue by StopMotionCuber** [ansible/ansible-lint#2779]
##### Summary

With `ansible-lint 6.9.1` the behavior of the `name` rule changed. It is assuming that a `|` is a prefix for a rule and therefore everything thereafter should follow the `name` rule.

This actually has 2 implications for us:
1. This behavior applies even though `name[prefix]` is disabled 
2. `name[prefix]` does not include checks for jinja templates.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.9.1 using ansible 2.14.0
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Have a sub-task in playbook with one of the following, and don't enable `name[prefix]`:

```console
- name: Latest links for {{ reposerver_centos_release_hps_subdirectories_links | join(', ') }}
```

```console
- name: Create string: 'user|adminlevel'
```

##### Desired Behavior

`name[casing]` should not trigger

##### Actual Behavior

`name[casing]` does trigger

##### Extra

It seems like this line [here](https://github.com/shatakshiiii/ansible-lint/blob/9e8abefadf8fdc1f56159b4266f5d189ef570ef0/src/ansiblelint/rules/name.py#L71) might be responsible for that behavior.

In case you need further information about this, let me know
